### DISCUSSION TITLE
DE10 - Lite support

### DISCUSSION THREAD

**Discussion by fpfrimer** [logisim-evolution/logisim-evolution#1075] [General]
Greetings, logisim evolution is a great contribution to the teaching of digital systems at my university. However, I wasn't able to understand how to add support for other cards. I didn't find this in the help topics. Is there any tutorial? I want to add terasic's [DE10 lite board](https://www.terasic.com.tw/cgi-bin/page/archive.pl?Language=English&No=1021)

**Comment by maehne** [comment]
@fpfrimer: Indeed, adding new FPGA boards is not yet described in the user documentation. Nevertheless, adding a new board is not too hard:

1. The first step is to open the "FPGA Board Editor" from the menu "FPGA".
2. Take a picture of your board in landscape mode, crop it, and save it in PNG format under a reasonable name, e.g., "DE10_lite.png"
3. Add the board image by clicking in the area with the message to do so.
4. Click on "Configure FPGA" and fill in the requested information. To do so, you'll need the board documentation and board schematics. FPGA family, part, package, and speed grade should be specified with the same names, which are used in Quartus Prime. For the pin locations, you have to use the very same names, which you would also use for the pin mapping in Quartus Prime.
5. Click and drag on the board image to draw rectangles around an I/O resource, which you would like to be able to map in Logisim-evolution. Select the type of the I/O resource and fill in the requested properties. Again, this has to match the information, you would use in Quartus Prime for the mapping.
6. Repeat 5. for each I/O resource.
7. Click on "Save board" to save the board configuration to an XML file.
8. Test your new board configuration in the "FPGA Commander" by choosing as "Target board" "Other" and open your board configuration file in the file chooser.
9. (Optional) Consider submitting your board configuration file for integration into Logisim-evolution.

If you have doubt how to enter some of the requested information, you may check out one of the built-in boards, which uses a similar kind of FPGA. @BFH-ktt1 can provide further hints regarding the board editor, as he implemented it.

@mbaillif and @BFH-ktt1: What do you think? Is the above short description of the FPGA board editor worth putting in our user documentation? Probably some screenshots would help illustrating the individual steps.

@MarcinOrlowski: I think you moved this too quickly to discussions, as support for a new board is kind of a feature request.

**Reply by MarcinOrlowski** [reply]
Well, perhaps I should hold my horses bit longer, but `Is there any tutorial? I want to add terasic's DE10 lite board` sounded less like FRE, more like open question, hence the move.

**Reply by fpfrimer** [reply]
@maehne Thank you very much. I will follow the steps as soon as possible. I will post the results here.

**Reply by Bahey-shalash** [reply]
Were you able to add the de10 lite to logism evolution?

**Comment by fpfrimer** [comment]
It was added in version 4.0.0

Felipe Pfrimer

COELE\UTFPR-TD
